toolip/fastapi_tools/config.py

```diff
@@ -1,35 +1,20 @@
 from typing import Optional
 
-from pydantic import BaseSettings, validator
+from pydantic import BaseSettings, root_validator
 
 
 class Configuration(BaseSettings):
-    doc_username: Optional[str] = None
-    doc_password: Optional[str] = None
+    doc_username: Optional[str] = ...  # type: ignore
+    doc_password: Optional[str] = ...  # type: ignore
 
     @property
     def doc_auth_is_on(self):
         return self.doc_username is not None and self.doc_password is not None
 
-    @validator('doc_username')
-    def username_not_empty(cls, value):
-        if value and len(value) > 0:
-            return value
-        raise ValueError('Username must not be an empty string')
-
-    @validator('doc_password')
-    def password_not_empty(cls, value, values):
-        username = values['doc_username']
-        if value and len(value) > 0:
-            if not username:
-                raise ValueError('Username and Password should both be set')
-            return value
-        raise ValueError('Password must not be an empty string')
-
-    class Config:
-        """Pydantic class to add prefix to properties defined."""
-
-        env_prefix = 'API_'  # defaults to no prefix, i.e. ""
-
-
-conf = Configuration()
+    @root_validator
+    def check_creds(cls, values):
+        username = values.get('doc_username')
+        password = values.get('doc_password')
+        if (username is None) ^ (password is None):
```

Comment:
The point is that you don't need the values, you just need to know if something was provided or not:
```suggestion
        if 'doc_username' in values ^ 'doc_password' in values:
```

Now you should have the default as `None` and not `...`